1
:6379
/home/gridsan/weizili/.conda/envs/flow/bin/python
Waiting for redis server at 172.31.2.51:6379 to respond...
Using IP address 172.31.2.51 for this node.
Process STDOUT and STDERR is being redirected to /state/partition1/slurm_tmp/433757.4294967294.0/logs.
Waiting for redis server at 127.0.0.1:6379 to respond...
Waiting for redis server at 127.0.0.1:18070 to respond...
Warning: Capping object memory store to 20.0GB. To increase this further, specify `object_store_memory` when calling ray.init() or ray start.
Starting the Plasma object store with 20.0 GB memory using /dev/shm.

======================================================================
View the web UI at http://localhost:8888/notebooks/ray_ui.ipynb?token=d7757efbb4a77e574820fcd714fb89dd92413444e4b0ff38
======================================================================

{'node_ip_address': '172.31.2.51', 'redis_address': '172.31.2.51:6379', 'object_store_addresses': ['/state/partition1/slurm_tmp/433757.4294967294.0/sockets/plasma_store'], 'raylet_socket_names': ['/state/partition1/slurm_tmp/433757.4294967294.0/sockets/raylet'], 'webui_url': 'http://localhost:8888/notebooks/ray_ui.ipynb?token=d7757efbb4a77e574820fcd714fb89dd92413444e4b0ff38'}

Started Ray on this node. You can add additional nodes to the cluster by calling

    ray start --redis-address 172.31.2.51:6379

from the node you wish to add. You can connect a driver to the cluster from Python by running

    import ray
    ray.init(redis_address="172.31.2.51:6379")

If you have trouble connecting from a different machine, check that your firewall is configured properly. If you wish to terminate the processes that have been started, run

    ray stop
Failed to connect to the redis server, retrying.
Waiting for redis server at 172.31.2.51:6379 to respond...
Using IP address 172.31.2.52 for this node.
Process STDOUT and STDERR is being redirected to out-temp-node-058/logs.
Warning: Capping object memory store to 20.0GB. To increase this further, specify `object_store_memory` when calling ray.init() or ray start.
Starting the Plasma object store with 20.0 GB memory using /dev/shm.
{'redis_address': '172.31.2.51:6379', 'node_ip_address': '172.31.2.52', 'redis_shards': ['172.31.2.51:18070'], 'object_store_addresses': ['out-temp-node-058/sockets/plasma_store'], 'raylet_socket_names': ['out-temp-node-058/sockets/raylet'], 'webui_url': ''}

Started Ray on this node. If you wish to terminate the processes that have been started, run

    ray stop
Traceback (most recent call last):
  File "../examples/rllib/stabilizing_the_ring.py", line 144, in <module>
    ray.init(num_cpus=N_CPUS + 1, redirect_output=False)
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/site-packages/ray/worker.py", line 1702, in init
    _internal_config=_internal_config)
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/site-packages/ray/worker.py", line 1444, in _init
    _internal_config=_internal_config)
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/site-packages/ray/services.py", line 1777, in start_ray_head
    _internal_config=_internal_config)
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/site-packages/ray/services.py", line 1391, in start_ray_processes
    get_logs_dir_path()))
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/site-packages/ray/tempfile_services.py", line 98, in get_logs_dir_path
    logs_dir = os.path.join(get_temp_root(), "logs")
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/site-packages/ray/tempfile_services.py", line 86, in get_temp_root
    try_to_create_directory(_temp_root)
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/site-packages/ray/tempfile_services.py", line 61, in try_to_create_directory
    raise e
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/site-packages/ray/tempfile_services.py", line 58, in try_to_create_directory
    os.makedirs(directory_path)
  File "/home/gridsan/weizili/.conda/envs/flow/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/tmp/ray/session_2019-08-30_23-51-37_26984'
